### ISSUE TITLE
Migrate deprecated Jaeger client

### ISSUE THREAD

**Issue by sophokles73** [eclipse/hono#3190]
The Jaeger client used in Hono has been deprecated and will be archived mid 2022 as [announced on the project website](https://www.jaegertracing.io/docs/1.32/client-libraries/). The client should be replaced with corresponding OpenTelemetry SDKs as described in the [migration guide](https://opentelemetry.io/docs/migration/opentracing/).

**Comment by calohmn** [comment]
Regarding the Jaeger support in Hono, I think there are 2 options.
We could use the `quarkus-opentelemetry-exporter-jaeger` Quarkus extension and thereby support directly sending the traces from the Hono components to the Jaeger collector. This doesn't require the OpenTelemetry Collector.
Or, we could use the `quarkus-opentelemetry-exporter-otlp` Quarkus extension and require usage of the OpenTelemetry Collector. The OpenTelemetry Collector can then be configured to export to the Jaeger Collector.

I guess the 2nd option, setting up the Hono containers each with an OpenTelemetry Collector sidecar (i.e. using the agent deployment method), would be the preferred deployment setup.
So, I would rather remove the explicit `jaeger` build option (maven profile) and focus just on the open telemetry support.
@sophokles73 WDYT?

**Comment by sophokles73** [comment]
>  guess the 2nd option, setting up the Hono containers each with an OpenTelemetry Collector sidecar (i.e. using the agent deployment method), would be the preferred deployment setup.

This seems to be analogous to how we deploy the Hono components in the Helm chart, right? However, for running the Hono integration tests, this sounds like we would need to start an additional container for the OpenTelemetry Collector, right?

**Comment by calohmn** [comment]
> However, for running the Hono integration tests, this sounds like we would need to start an additional container for the OpenTelemetry Collector, right?

Yes, but I think this is manageable. The `quarkus-opentelemetry-exporter-jaeger` extension is anyway being phased out of the main Quarkus repo at some point (see https://github.com/quarkusio/quarkus/issues/17771).

**Comment by sophokles73** [comment]
Ok, then let's go for the *quarkus-opentelemetry-exporter-otlp* ...

**Comment by sophokles73** [comment]
@calohmn can this be closed?
